File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    int r;
    int i;
    do
    {
        height = get_int("height: ");
        for (r = 0; r < height; r++)
        {
            for (i=height-r-1; i > 0; i--)
            {
                printf(" ");
            }
            for (i=0; i < r+2; i++)
            {
                printf("%c", '#');
            }
        printf("\n");
        }
    }while (height > 23 || height < 0);
}

File: mario.py
from cs50 import get_int
def main():
    height = get_int("height: ")
    checkHeight(height)

def checkHeight(height):
    if height < 24 and height >= 0:
        mario(height)
    else:
        main()

def mario(height):
    for r in range(height):
        for i in range(0, height-r-1):
            print(" ", end="")
        for j in range(0, r+2):
            print('#', end="")
        print("")

if __name__ == "__main__":
    main()